src/app/GitUI/Editor/ComboBoxWatermarkManager.cs

```diff
@@ -0,0 +1,185 @@
+namespace GitUI.Editor;
+
+/// <summary>
+///  Manages watermark text display for a ComboBox control.
+/// </summary>
+internal sealed class ComboBoxWatermarkManager : IDisposable
+{
+    private readonly ComboBox _comboBox;
+    private readonly string _watermarkText;
+    private Font _watermarkFont;
+    private readonly Color _watermarkColor = SystemColors.GrayText;
+    private Font _originalFont;
+    private Color _originalForeColor;
+    private bool _isWatermarkActive;
+    private bool _isDisposed;
+
+    /// <param name="comboBox">The ComboBox control to manage.</param>
+    /// <param name="watermarkText">The watermark text to display.</param>
+    public ComboBoxWatermarkManager(ComboBox comboBox, string watermarkText)
+    {
+        _comboBox = comboBox ?? throw new ArgumentNullException(nameof(comboBox));
+        _watermarkText = watermarkText ?? string.Empty;
+
+        _originalFont = _comboBox.Font;
+        _originalForeColor = _comboBox.ForeColor;
+        _watermarkFont = new Font(_originalFont, FontStyle.Italic);
+
+        if (string.IsNullOrEmpty(_comboBox.Text))
+        {
+            ShowWatermark();
+        }
+
+        _comboBox.Enter += ComboBox_Enter;
+        _comboBox.LostFocus += ComboBox_LostFocus;
+        _comboBox.TextChanged += ComboBox_TextChanged; // Handle programmatic changes or pasting
+        _comboBox.FontChanged += ComboBox_FontChanged; // Keep original font updated
+        _comboBox.ForeColorChanged += ComboBox_ForeColorChanged; // Keep original color updated
```

Comment:
These comments are typical for an AI (explaining stuff for newbies - which helps assessing the AI answer).
Such obvious things do not need to be kept in the code IMO.